src/views/SafeAreaView.js

```diff
@@ -195,6 +204,21 @@ class SafeView extends Component {
       });
     }
 
+    // new height/width should only include padding from insets
+    // height/width should not be affected by padding from style obj
+    if (style.height && typeof style.height === 'number') {
```

Comment:
confused about what is going on here and below

Reply:
when height is specified on the SafeAreaView, height is supposed to include padding, so the inset padding makes the view much smaller than it's supposed to be. This is the problem I was having with the header. Making the height include the inset padding fixes this, but also doesn't break the expectation that the specified padding from style does not make the component larger than that height.

Reply:
also, height can be a string for percent. This doesn't handle that case.